variants/frontend-stimulus/template.rb

```diff
@@ -0,0 +1,96 @@
+source_paths.unshift(File.dirname(__FILE__))
+
+puts "Setting up stimulus.js"
+
+yarn_add_dependencies %w[
+  @hotwired/stimulus
+  @hotwired/stimulus-webpack-helpers
+]
+
+if TEMPLATE_CONFIG.use_typescript?
+  copy_file "variants/frontend-stimulus/hello_controller.ts", "app/frontend/stimulus/controllers/hello_controller.ts"
+else
+  copy_file "variants/frontend-stimulus/hello_controller.js", "app/frontend/stimulus/controllers/hello_controller.js"
+end
+
+##
+# Rename application.js to applicaiton.ts if it hasn't already been done by some
+# other variant.
+#
+app_js_path = "app/frontend/packs/application.js"
+app_ts_path = "app/frontend/packs/application.ts"
+FileUtils.mv(app_js_path, app_ts_path) if TEMPLATE_CONFIG.use_typescript? && File.exist?(app_js_path)
```

Comment:
🐘 I think we could easily get away with using a `js_ext` variable, which should let you half the this file

Reply:
Yeah I'm planning to sort the duplication once things settle. I wasn't sure how much the TS would diverge so held off